tests/integration/test_manifest_validation.py

```diff
@@ -0,0 +1,582 @@
+                "mock_manifests/TestManifestValidation_test_manifest_validation_basic_valid.csv",
+            ),
+        ],
+    )
+    def test_manifest_validation_basic_valid(
+        self,
+        input_data_type: str,
+        input_file_name: str,
+        flask_client: FlaskClient,
+        request_headers: Dict[str, str],
+        testing_config: ConfigurationForTesting,
+        helpers: Helpers,
+    ) -> None:
+        """
+        Test that the manifest validation API returns no errors when a valid manifest is provided.
+
+        We are validating the following:
+
+        When we upload a valid manifest file to the /v1/model/validate endpoint...
+        - The response is successful (Response Code 200)
+        - The response content includes an 'errors' and 'warnings' key
+        - There are NO error values for the 'errors' key
+        - There are NO warnings values for the 'warnings' key
+
+        """
+        # GIVEN the manifest validation endpoint and parameters
+        url = f"{testing_config.schematic_api_server_url}/v1/model/validate"
+        params = {
+            "schema_url": EXAMPLE_SCHEMA_URL,
+            "data_type": input_data_type,
+            "data_model_labels": "class_label",
+            "restrict_rules": False,
+        }
+
+        # AND a valid file
+        file_path = helpers.get_data_path(input_file_name)
+
+        # WHEN we make a POST request to validate the file
+        response = (
+            requests.post(
+                url,
+                params=params,
+                files={"file_name": open(file_path, "rb")},
+                headers=request_headers,
+            )
+            if testing_config.use_deployed_schematic_api_server
+            else flask_client.post(
+                url,
+                headers=request_headers,
+                query_string=params,
+                data={"file_name": open(file_path, "rb")},
+            )
+        )
+
+        # THEN we expect a successful response
+        assert (
+            response.status_code == 200
+        ), f"Got status code: {response.status_code}. Expected '200'."
+
+        # AND with expected keys in the json
+        response_json = (
+            response.json()
+            if testing_config.use_deployed_schematic_api_server
+            else response.json
+        )
+        assert "warnings" in response_json.keys()
+        assert "errors" in response_json.keys()
+
+        # AND with no expected errors
+        assert len(response_json.get("errors")) == 0
+
+        # AND with no expected warnings
+        assert len(response_json.get("warnings")) == 0
+
+    @pytest.mark.local_or_remote_api
+    @pytest.mark.parametrize(
+        ("input_data_type", "input_file_name"),
+        [
+            (
+                "Patient",
+                "mock_manifests/TestManifestValidation_test_patient_manifest_invalid.csv",
+            ),
+        ],
+    )
+    def test_manifest_validation_basic_invalid(
```

Comment:
Any particular reason for this to be parameterized with an input of length 1?

Reply:
@jaymedina ^

---
I like the idea that we should be able to throw another file into these tests if we wanted to expand on them, but I'm not sure if that was the original intention on this.

Reply:
Yes, the original intention was to have a "framework" for future devs to just add into the existing parametrization.